### DISCUSSION TITLE
up-show-for :blank on select2 field

### DISCUSSION THREAD

**Discussion by thinkwelltwd** [unpoly/unpoly#419] [Q&A] [answered]
I have a [select2 ajax field](https://select2.org/data-sources/ajax) that's marked with `id="company" up-switch=".target_css"`, and the target element is set to `up-hide-for=":blank"`. 

When the form initially loads with and the select2 field is blank, the target element is hidden and selecting a `company` value displays the `target_css` field as expected. 

However, when clearing the `company` field, the target field doesn't become hidden. I've tried clearing with the "x" control, as well as programmatically with:

```javascript
$('#company').val(null).trigger('change')
```

But the target field still doesn't disappear. If I change the `up-hide-for` value to a real value, such as `up-hide-for="2150"`, then the field hides & displays as expected **when cycling to & from 2150 and some other**. However, when changing from 2150 to a null value, the `target_css`  field does not re-appear.

Is there a way to programmatically clear the select2 `up-switch` field such that the `up-hide-for` field notices? Or is there an event that I can fire to make the `up-hide-for` field respond?

Thanks!

**Comment by thinkwelltwd** [comment, marked as answer]
:man_facepalming:  I discovered my problem; clearing the select2 field emits the `select2:unselect` event and I needed have unpoly emit a change event when that occurs.

The snippet below got everything working; hope this helps someone. :blush:  

```javascript
$element.on('select2:select select2:unselect', function (e) {
  up.emit($element, 'change');
});
```

**Comment by triskweline** [comment]
The background is that [select2 emits synthetic jQuery events](https://makandracards.com/makandra/71890-events-triggered-by-jquery-cannot-be-observed-by-native-event-listeners). So select2 already emits a synthetic `change` event, but it cannot be observed by native event listeners like Unpoly.

My team has migrated newer projects to [select2 alternatives without jQuery](https://makandracards.com/makandra/481570-select2-alternatives-without-jquery).
